### PR TITLE
Add create document api endpoint

### PR DESCRIPTION
**Pull request by sampaccoud**
## Purpose

We want trusted external applications to be able to create documents via the API on behalf of any user. 

## Proposal

Add an action API endpoint limited to document creation and with a basic server-to-server authentication: 
`/api/v1.0/documents/create-for-user/`

Expected payload:
```python
{
    "title": "My document title",     # document title 
    "content": "My document content", # document content in ydoc format
    "sub": "123",                     # "sub" of the user in the remote application who may pre-exist in our db
    "email": "john.doe@example.com",  # email to send invitation in case the user "sub" doesn't exist in our db
    "language": "fr-fr",              # [optional] language in which the invitation email should be sent
    "message": "My custom message",   # [optional] custom message in the invitation email
    "subject": "My custom subject",   # [optional] custom subject for the invitation email
}
```
The user may or may not pre-exist in our database and should be notified of the document creation by email.

### CODE REVIEW

File: src/backend/core/api/serializers.py [hunk 1]
```diff
@@ -227,6 +228,83 @@ def validate_id(self, value):
...
+        except (models.User.DoesNotExist, KeyError):
+            user = None
+            email = validated_data["email"]
+        else:
+            email = user.email
+            language = user.language or language
+
+        document = models.Document.objects.create(
+            title=validated_data["title"],
+            content=validated_data["content"],
```
Comment (lebaudantoine) [inline src/backend/core/api/serializers.py:269]: people won't be able to return yjs string. You should at least convert the string received in yjs format?

File: src/backend/core/api/serializers.py [hunk 2]
```diff
@@ -227,6 +228,83 @@ def validate_id(self, value):
...
+    Both "sub" and "email" are required because the external app calling doesn't know
+    if the user will pre-exist in Docs database. If the user pre-exist, we will ignore the
+    submitted "email" field and use the email address set on the user account in our database
+    """
+
+    # Document
+    title = serializers.CharField(required=True)
+    content = serializers.CharField(required=True)
+    # User
+    sub = serializers.CharField(required=False)
```
[the remark this answers is not in the document]
Comment (lebaudantoine) [reply src/backend/core/api/serializers.py:244]: Fixed, I've updated unit tests.

**Review by lebaudantoine — COMMENTED** [review]
This endpoint breaks encapsulation by requiring external services to pass data in the highly specific Yjs format, which is tightly coupled to the Doc implementation. It impacts the usability of the endpoint.

Moreover, it should implement a robust validation to ensure the content value is well-formed and will not cause errors when opening the document.

Finally, allowing the conversion of Markdown strings into Yjs format, supporting broader use cases like importing or transforming existing documents, would be a really nice enhancement.

Some of these requirements depend on the ability to execute JavaScript code. This capability is expected to be available soon, once @AntoLC completes the upgrade to the collaboration service.

This PR also introduces documents without any owner, which might need to be discussed.

File: src/backend/core/services/converter_services.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    @property
+    def auth_header(self):
+        """Build microservice authentication header."""
+        return f"Bearer {settings.CONVERSION_API_KEY}"
+
+    def convert(self, text):
+        """Convert a Markdown text into our internal format using an external microservice."""
+
+        if text == "":
```
Comment (sampaccoud) [inline src/backend/core/services/converter_services.py:39]: If not text?

**Review by sampaccoud — COMMENTED** [review]
All nitpicks. Move ahead!

File: src/backend/core/api/serializers.py [hunk 4]
```diff
@@ -266,9 +274,14 @@ def create(self, validated_data):
             email = user.email
             language = user.language or language
 
+        try:
+            content = ConverterService().convert(text=validated_data["content"])
```
[the remark this answers is not in the document]
Comment (lebaudantoine) [reply src/backend/core/api/serializers.py:278]: I took `YdocConverter`

File: src/backend/core/tests/documents/test_api_documents_create_for_owner.py [hunk 5]
```diff
@@ -80,6 +80,48 @@ def test_api_documents_create_for_owner_authenticated_forbidden():
...
+    }
+
+    response = APIClient().post(
+        "/api/v1.0/documents/create-for-owner/",
+        data,
+        format="json",
+        HTTP_AUTHORIZATION="Bearer DummyToken",
+    )
+
+    assert response.status_code == 400
```
[the remark this answers is not in the document]
Comment (lebaudantoine) [reply src/backend/core/tests/documents/test_api_documents_create_for_owner.py:100]: I've simplify followed the same logic as your initial tests!

**Pull request merged**